pasqal_cloud/utils/filters.py

```diff
@@ -0,0 +1,122 @@
+from datetime import datetime
+from typing import Any, Dict, List, Optional, Union
+from uuid import UUID
+
+from pydantic import BaseModel, field_serializer, field_validator
+
+from pasqal_cloud.utils.constants import JobStatus
+
+
+class BaseFilters(BaseModel):
+    @staticmethod
+    def convert_to_list(value: Any) -> Any:
+        if not value:
+            return None
+        if isinstance(value, list):
+            return value
+        return [value]
+
+    @staticmethod
+    def convert_str_to_uuid(value: Any) -> Any:
+        if isinstance(value, str):
+            try:
+                return UUID(value)
+            except ValueError:
+                raise ValueError(f"Cannot convert {value} to UUID.")
+        return value
+
+    @field_serializer("start_date", "end_date", check_fields=False)
+    def serialize_datetime_to_isoformat(self, date: datetime) -> str:
+        return date.isoformat()
+
+
+class RebatchFilters(BaseFilters):
+    """
+    This class is used for providing the filters considering data related to the
+    jobs within the batch to re-create.
+
+    The filtering logic employs an AND logic between different fields,
+    meaning all conditions must be met simultaneously for a result to be included.
+
+    However, within fields that accept lists, each item in the list is treated as
+    a possibility, using an OR logic within that field.
+
+    Attributes:
+        id: Filter by job ID(s), it can be a single UUID/str or a list of UUIDs/strs.
+        status: Filter by job status(es),
+        it can be a single JobStatus or a list of JobStatuses.
+        min_runs: Minimum number of runs.
+        max_runs: Maximum number of runs.
+        start_date: Filter by start date.
+        end_date: Filter by end date.
+    """
+
+    id: Optional[Union[List[Union[UUID, str]], Union[UUID, str]]] = None
```

Comment:
```suggestion
    job_ids: Optional[Union[List[Union[UUID, str]] = None
```

As discussed in our previous PP session, I prefer not to combine a list and a single value in the same attribute

Reply:
I am in favor of giving the possibility to the user to choose as it is the most natural way for the users to filter what it needs. So I would keep it like this.

Then we can handle the data given and adapt it to the API if it only expects lists or single items.